packages/notification-center/src/components/novu-provider/NovuProvider.tsx

```diff
@@ -94,42 +99,41 @@ export function NovuProvider({
   const logout = useCallback(() => {
     removeToken(apiService);
     disconnectSocket();
-    setSessionInitialized(false);
-  }, [removeToken, disconnectSocket, apiService]);
+    setSessionInfo((old) => ({ ...old, isSessionInitialized: false }));
+  }, [setSessionInfo, disconnectSocket, apiService]);
 
   const contextValue = useMemo(
     () => ({
       backendUrl,
       socketUrl,
-      applicationIdentifier,
-      subscriberId,
-      subscriberHash,
-      isSessionInitialized,
+      applicationIdentifier: sessionInfo.applicationIdentifier,
+      subscriberId: sessionInfo.subscriberId,
+      subscriberHash: sessionInfo.subscriberHash,
+      isSessionInitialized: sessionInfo.isSessionInitialized,
       apiService,
       socket,
       fetchingStrategy,
       setFetchingStrategy,
       onLoad,
       logout,
     }),
-    [
-      backendUrl,
-      socketUrl,
-      applicationIdentifier,
-      subscriberId,
-      subscriberHash,
-      isSessionInitialized,
-      apiService,
-      socket,
-      fetchingStrategy,
-      setFetchingStrategy,
-      onLoad,
-      logout,
-    ]
+    [backendUrl, socketUrl, sessionInfo, apiService, socket, fetchingStrategy, setFetchingStrategy, onLoad, logout]
   );
 
   useEffect(() => disconnectSocket, [disconnectSocket]);
 
+  useEffect(() => {
+    setSessionInfo((old) => ({
+      ...old,
+      isSessionInitialized: false,
+      applicationIdentifier,
+      subscriberId,
+      subscriberHash,
+    }));
+    logout();
+    queryClient.refetchQueries([...SESSION_QUERY_KEY, applicationIdentifier, subscriberId, subscriberHash]);
+  }, [logout, subscriberId, applicationIdentifier, subscriberHash]);
```

Comment:
when the props `subscriberId, applicationIdentifier, subscriberHash` are changing, set the session to `isSessionInitialized: false` and then reinitialize the session

Reply:
Do you mean that the reinitialize will be in onSuccessfulSession?